pkg/kubernetessecrets/kubernetes_secrets_builder.go

```diff
@@ -69,15 +67,15 @@ func CreateKubernetesSecretFromItem(kubeClient kubernetesClient.Client, secretNa
 	currentAnnotations := currentSecret.Annotations
 	currentLabels := currentSecret.Labels
 	currentSecretType := string(currentSecret.Type)
-	if currentSecretType == "" {
-		currentSecretType = "Opaque"
+	if !reflect.DeepEqual(currentSecretType, secretType) {
+		return errs.New("Cannot change secret type. Secret type is immutable")
```

Comment:
Shouldn't we do a create + delete in this case?

Reply:
Ahh right, if this is expected behavior in K8s, let's definitely stick with that :+1: 

I think I confused with Terraform behavior for a bit.